### PR TITLE
[TS] LPS-167991

### PR DESCRIPTION
**Pull request by crodriguezy**
Hello team, this PR fixes case [LPS-167991](https://issues.liferay.com/browse/LPS-167991), which was originally reported as a 7.2.x-only fix.  

But after discussing it with @antonio-ortega and @dacousalr in [PTR-3478](https://issues.liferay.com/browse/PTR-3478), we realized that this could also be a potential problem in master, and the best way to fix it globally was adding the `escape=true` attribute to the _Message Tag_ when it is used to build titles.

Please let me know your thoughts about it.

Thank you!

### CODE REVIEW

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
Hi @liferay-frontend , before I fix the SF failures, I wonder if these changes look ok to you,
Thank you!

**Comment by crodriguezy** [comment]
Hi @liferay-frontend , did you have the time to take a look at these changes?
I will fix the SF once I get your approval.
Thanks

**Review by markocikos — COMMENTED** [review]
@crodriguezy Sorry for the late response! It is very hard to make a review, since there are a lot of fake SF changes. Please fix the conflict, fix SF, and force-push.

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
Hi @markocikos , I've finally managed to fix the SF (it may have included extra files where the escape attribute is not added, but I can exclude those later if needed).
Could you please take a look at it now?
Thanks

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/scope.jsp [hunk 1]
```diff
@@ -72,48 +73,127 @@ List<Group> selectedGroups = editAssetListDisplayContext.getSelectedGroups();
...
+		var groupIds = document.getElementById('<portlet:namespace />groupIds');
+
+		if (groupIds) {
+			var searchContainerData = searchContainer.getData();
+
+			groupIds.setAttribute('value', searchContainerData.split(','));
+
+			<portlet:namespace />saveSelectBoxes();
+		}
+	}
```
Comment (markocikos) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/scope.jsp:198]: Are all these changes in `asset-list-web` related to the [ticket](https://issues.liferay.com/browse/LPS-167991)?

**Review by markocikos — COMMENTED** [review]
@crodriguezy 
1. See comment inline about changes in `asset-list-web`.
2. Searching DXP for `'<liferay-ui:message` in `*.jsp*`, I get 415 results in 155 files. In this PR, we are modifying ~50. Is there a reason why we are modifying some but not others?
3. Looking at source code, it seems we already have a mechanism that transforms labels to unicode characters, see screenshot from `master` below. Are there some cases where this would not be enough?
screen

**Comment by crodriguezy** [comment]
Hi @markocikos ,

Answering your questions:

1. No, it is something the rebase did, I'll look into it.
2. We decided the best way to fix it globally was adding the escape=true attribute to the Message Tag when it is used to build **titles**, I'm not sure if you filtered your search to only find those title building ocurrences.
3. I didn't check whether this could happen in all places where we build titles because there are a lot. It is certainly not enough for the case described in [LPS-167991](https://issues.liferay.com/browse/LPS-167991), but if you feel we can skip some of them, please let me know.

**Comment by crodriguezy** [comment]
Hi @markocikos ,

Just a quick question, how did you try to reproduce the bug in master? 

In the LPS description the listed steps were only valid in 7.2.x:

```
Steps to reproduce in 7.2.x:

Add a widget page with an Asset Publisher.
Change URL language to /fr/
Select AP configuration -> Asset Selection -> Ensemble de Contenus
Click on 'Selectionner L'Ensemble de Contenus' to display the options.
Click on button 'Sélectionner'
```

In master, since the title/message resolves to 'Collections' (without any single quote included) there is no way to reproduce it:

`Notes: Although the steps to reproduce only apply to 7.2.x, in master/7.3.x it could be also a problem if any translation for titles contains a special character (typically an apostrophe) that is not escaped and leads to a js SyntaxError. That's the reason we decided to contribute the solution to master as well.`

**Comment by markocikos** [comment]
> Just a quick question, how did you try to reproduce the bug in master?

As you said, we cannot replicate it directly. What I did is to replace `select-collection` on [this line](https://github.com/liferay/liferay-portal/blob/master/modules/apps/asset/asset-publisher-web/src/main/resources/META-INF/resources/configuration/asset_list.jsp#L92) with `administrators-email`. It can be any string that contains an apostrophe. This is what I get, including debugging of encoding on frontend side:

https://user-images.githubusercontent.com/5435511/204782691-4fe84d18-4d67-44e8-8f81-526d9e8375e5.mp4

**Comment by eltonccdantas** [comment]
Since this branch has some conflicts to be resolved, I'll wait to analyze it with a QA perspective.

**Comment by eltonccdantas** [comment]
Since this branch has some conflicts to be resolved, I'll wait to analyze it from a QA perspective.

**Comment by crodriguezy** [comment]
Hi @eltonccdantas , this PR will most likely be modified, but I still need to analyze what Marko suggested in his last comment

**Comment by crodriguezy** [comment]
Hi @markocikos , first of all excuse me for my delayed response. 

Thanks to your analysis I was able to identify the root problem of the JS context not being set in 7.2.x. We were not using the <aui:script> tag, where we set the JAVASCRIPT_CONTEXT attribute to true, in asset_list.jsp (in master we do).

Please see this new PR where I've used it and now the JS error is gone and the selection becomes usable: https://github.com/liferay/liferay-portal-ee/pull/29396

**Pull request closed without merging**